tests/CatalogTests/CatalogIndexReaderTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using NgTests.Data;
+using NuGet.Services.Metadata.Catalog;
+using Xunit;
+
+namespace CatalogTests
+{
+    public class CatalogIndexReaderTests
+    {
+        [Fact]
+        public async Task GetEntries()
+        {
+            // Arrange
+            var indexUri = "http://tempuri.org/index.json";
+            var responses = new Dictionary<string, string>()
+            {
+                { "http://tempuri.org/index.json", TestCatalogEntries.TestCatalogStorageWithThreePackagesIndex },
+                { "http://tempuri.org/page0.json", TestCatalogEntries.TestCatalogStorageWithThreePackagesPage },
+            };
+
+            var reader = new CatalogIndexReader(
+                new Uri(indexUri),
+                new CollectorHttpClient(new InMemoryHttpHandler(responses)));
+
+            // Act
+            var entries = await reader.GetEntries();
+
+            // Assert
+            var entryList = entries.ToList();
+            Assert.Equal(3, entryList.Count);
+
+            Assert.Equal("http://tempuri.org/data/2015.10.12.10.08.55/listedpackage.1.0.1.json", entryList[0].Uri.ToString());
```

Comment:
Add verification of all the fields to this test. You're missing `Types`, `CommitId`, and `Version`.